### PR TITLE
Developer Certificate of Origin workflow

### PR DESCRIPTION
**Pull request by peternied**
### Issues Resolved
* Resolves https://github.com/opensearch-project/opensearch-build/issues/619
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]

**Comment by dblock** [comment]
Let's add enabling this workflow in https://github.com/opensearch-project/.github/blob/d32533a0ebac56c4ece38a0ec89f1264cf5971fc/MAINTAINERS.md?

File: MAINTAINERS.md [hunk 1]
```diff
@@ -57,6 +57,14 @@ Respond to enhancement requests, and forum posts. Allocate time to reviewing and
 
 Keep the `main` branch at production quality at all times. Backport features as needed. Cut release branches and tags to enable future patches. 
 
+### Add Continious Integration Checks
```
Comment (dblock) [inline MAINTAINERS.md:60]: Add to TOC

File: MAINTAINERS.md [hunk 2]
```diff
@@ -57,6 +57,14 @@ Respond to enhancement requests, and forum posts. Allocate time to reviewing and
...
+
+Add integration checks that validate pull requests and pushes to ease the burden on Pull Request reviewers
```
Comment (dblock) [inline MAINTAINERS.md:62]: Add a period.

File: MAINTAINERS.md [hunk 3]
```diff
@@ -57,6 +57,14 @@ Respond to enhancement requests, and forum posts. Allocate time to reviewing and
...
+
+#### Developer Certificate of Origin workflow
```
Comment (dblock) [inline MAINTAINERS.md:64]: Capitalize Workflow

**Review by dblock — CHANGES_REQUESTED** [review]

File: MAINTAINERS.md (hunk 1 above)
Comment (peternied) [reply MAINTAINERS.md:64]: Found this tool, http://ecotrust-canada.github.io/markdown-toc/ any concern about including its comment?  IMO its a great way for future writers to regenerate a consistent TOC

Comment (dblock) [reply MAINTAINERS.md:64]: I use a [Markdown All in One plugin for VSCode](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) and never have to think about it.

File: MAINTAINERS.md [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
-  - [Prioritize Security](#prioritize-security)
-  - [Review Pull Requests](#review-pull-requests)
-  - [Triage Open Issues](#triage-open-issues)
-  - [Be Responsive](#be-responsive)
-  - [Maintain Overall Health of the Repo](#maintain-overall-health-of-the-repo)
-  - [Use Semver](#use-semver)
-  - [Release Frequently](#release-frequently)
-  - [Promote Other Maintainers](#promote-other-maintainers)
-  - [Describe the Repo](#describe-the-repo)
+  * [Uphold Code of Conduct](#uphold-code-of-conduct)
```
Comment (dblock) [inline MAINTAINERS.md:4]: This rewrites formatting using different style bullets which isn't necessary for GitHub and will be overwritten by other plugins, can we not?

Comment (peternied) [reply MAINTAINERS.md:4]: switching to your plugin

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
I think this is obsolete now since we have an app, @peternied should we kill it?

**Comment by peternied** [comment]
I think we should kill it, the app is far better

**Comment by dblock** [comment]
> I think we should kill it, the app is far better

https://github.com/opensearch-project/.github/pull/112

**Pull request merged**